src/core/blocks/strings.cpp

```diff
@@ -133,6 +133,55 @@ struct Replace : public Common {
   }
 };
 
+struct Join {
+  static CBTypesInfo inputTypes() { return CoreInfo::StringSeqType; }
+  static CBTypesInfo outputTypes() { return CoreInfo::StringType; }
+
+  static CBParametersInfo parameters() { return CBParametersInfo(params); }
+
+  void setParam(int index, const CBVar &value) {
+    switch (index) {
+    case 0:
+      _separator = value.payload.stringValue;
+      break;
+    default:
+      break;
+    }
+  }
+
+  CBVar getParam(int index) {
+    switch (index) {
+    case 0:
+      return Var(_separator);
+    default:
+      return Var::Empty;
+    }
+  }
+
+  CBVar activate(CBContext *context, const CBVar &input) {
+    if (input.payload.seqValue.len == 0)
+      return Var::Empty; // FIXME or Var("")?
+
+    std::ostringstream ss;
+    ss << input.payload.seqValue.elements[0].payload.stringValue;
```

Comment:
Many allocations here...
Can you instead add a `std::string _buffer` inside the class
and here `_buffer.clear()`
followed by `_buffer.append(x)`
This will allow to share memory 100% as clear won't free()
This is the chainblocks spirit :)

Reply:
Please check `(AppendTo)` for example